lib/src/chargebee.dart

```diff
@@ -75,19 +75,24 @@ class Chargebee {
   /// [product] product object to be passed.
   ///
   /// [customerId] it can be optional.
+  /// [customer] it can be optional.
   /// if passed, the subscription will be created by using customerId in chargebee.
   /// if not passed, the value of customerId is same as SubscriptionId.
   ///
   /// If purchase success [PurchaseResult] object be returned.
   /// Throws an [PlatformException] in case of failure.
   static Future<PurchaseResult> purchaseProduct(
-    Product product, [
+    Product product, {
+    @Deprecated(
+      'This param will be removed in upcoming release, Use CBCustomer object to pass customer info instead',
+    )
     String? customerId = '',
-  ]) async {
-    customerId ??= '';
+    CBCustomer? customer,
+  }) async {
+    final map = _convertObjectToMap(product, customerId, customer);
```

Comment:
Why do we need to send both customerId and customer object to the same method?
If we want to support both customerId and customer object, lets create separate methods for both.

Reply:
Wouldn't both of these be optional param in this case? 🤔